settings/src/components/backup-codes.js

```diff
@@ -14,13 +14,26 @@ import { refreshRecord } from '../utilities';
 
 /**
  * Setup and manage backup codes.
+ *
+ * @param props
+ * @param props.setScreen
  */
-export default function BackupCodes() {
+export default function BackupCodes( { setScreen } ) {
 	const {
-		user: { backupCodesEnabled },
+		user: { totpEnabled, backupCodesEnabled },
 	} = useContext( GlobalContext );
 	const [ regenerating, setRegenerating ] = useState( false );
 
+	// If TOTP hasn't been enabled, the user should not have access to BackupCodes component.
+	// This is primarily added to prevent users from accessing through the URL.
+	if ( ! totpEnabled ) {
+		const currentUrl = new URL( document.location.href );
+		currentUrl.searchParams.set( 'screen', 'account-status' );
+		window.history.pushState( {}, '', currentUrl );
+		setScreen( 'account-status' );
```

Comment:
I actually had a go at doing just this recently and my approach was to reuse the internals of the `clickScreenLink` function, as this is almost exactly that. I created a branch which changed that to a basic navigation handler: https://github.com/WordPress/wporg-two-factor/compare/fix/unauthorized-backup-codes?expand=1

Events are handled at the component level and `navigateToScreen` is called.

See what you think.

Reply:
I like the enhancement 👍 Thanks for it.
I've cherry-picked the commits from your branch and tested the behavior on a sandbox, things worked as expected to me. Do you mind taking another look and reviewing again? Thanks.